src/Shared/QueryStringEnumerable.cs

```diff
@@ -0,0 +1,202 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Buffers;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+#if QueryStringEnumerable_In_WebUtilities
+namespace Microsoft.AspNetCore.WebUtilities
+#else
+namespace Microsoft.AspNetCore.Internal
+#endif
+{
+    /// <summary>
+    /// An enumerable that can supply the name/value pairs from a URI query string.
+    /// </summary>
+    public readonly ref struct QueryStringEnumerable
```

Comment:
Normally in shared source the type is public in at most one instance. Do we really want the same type to be public in multiple assemblies and namespaces? That seems really confusing, especially if someone pulls both dependencies into their project.

Reply:
Good point! I meant to toggle the top-level access modifier based on whether it's internal. Done this in a further commit now.